src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,213 @@
+
+// Update accumulated data
+void IntegerHistogram1D::updateAccumulatedData()
+{
+    std::pair<Data1D, std::optional<int>> average = createDisplayData();
+    accumulatedData_ = average.first;
+    auto minBin = average.second.value_or(0);
+    // Poke bin values and errors into array
+    for (const auto &[key, value] : averages_)
+    {
+        auto n = key - minBin;
+        accumulatedData_.value(n) = value.value();
+        accumulatedData_.error(n) = value.stDev();
+    }
+}
+
+// Create dsiplay data
+const std::pair<Data1D, std::optional<int>> IntegerHistogram1D::createDisplayData()
+{
+
+    // Get limiting key values
+    std::optional<int> expectedMinimum = raw_.empty() ? std::nullopt : std::optional<int>(raw_.begin()->first);
+    if (minimum_)
+        expectedMinimum = expectedMinimum ? std::min(*minimum_, *expectedMinimum) : *minimum_;
+
+    std::optional<int> expectedMaximum = raw_.empty() ? std::nullopt : std::optional<int>(std::prev(raw_.end())->first);
+    if (maximum_)
+        expectedMaximum = expectedMaximum ? std::max(*maximum_, *expectedMaximum) : *maximum_;
+
+    // Need to check on null values - set to zeroes if both null, or if one is null set it to match the other
+    if (!expectedMinimum && !expectedMaximum)
+    {
+        expectedMinimum = 0;
+        expectedMaximum = 0;
+    }
+    else if (!expectedMinimum)
+        expectedMinimum = expectedMaximum;
+    else if (!expectedMaximum)
+        expectedMaximum = expectedMinimum;
+
+    auto expectedNBins = (*expectedMaximum - *expectedMinimum) + 1;
+
+    // Set up data
+    Data1D data;
+    data.initialise(expectedNBins, true);
+    auto x = *expectedMinimum;
+    for (auto n = 0; n < expectedNBins; ++n)
+        data.xAxis(n) = x++;
+
+    return std::make_pair(data, expectedMinimum);
+}
+
+// Initialise with specified bin range
+void IntegerHistogram1D::initialise(std::optional<int> xMin, std::optional<int> xMax)
+{
+    clear();
+
+    minimum_ = xMin;
+    maximum_ = xMax;
+
+    raw_.clear();
+    averages_.clear();
+}
+
+// Zero histogram bins
+void IntegerHistogram1D::zeroBins()
+{
+    raw_.clear();
+    averages_.clear();
+    nBinned_ = 0;
+    nMissed_ = 0;
+}
+
+// Return minimum value for data
+std::optional<int> IntegerHistogram1D::minimum() const { return minimum_; }
+
+// Return maximum value for data
+std::optional<int> IntegerHistogram1D::maximum() const { return maximum_; }
+
+// Bin specified value, returning success
+bool IntegerHistogram1D::bin(int x)
+{
+    if ((minimum_ && x < *minimum_) || (maximum_ && x > *maximum_))
+    {
+        nMissed_++;
+        return false;
+    }
+
+    raw_[x]++;
+    ++nBinned_;
+
+    return true;
+}
+
+// Return number of values binned over all bins
+long int IntegerHistogram1D::nBinned() const { return nBinned_; }
+
+// Accumulate current histogram bins into averages
+void IntegerHistogram1D::accumulate()
+{
+    for (auto &[key, value] : raw_)
+        averages_[key] += (SampledDouble)raw_[key];
```

Comment:
Small thing, but we can convert to `double` here instead of `SampledDouble` to keep things simple.
```suggestion
        averages_[key] += (double)raw_[key];
```